cli/command/swarm/opts.go

```diff
@@ -24,13 +24,17 @@ const (
 	flagToken               = "token"
 	flagTaskHistoryLimit    = "task-history-limit"
 	flagExternalCA          = "external-ca"
+	flagKeepOldSnapshots    = "keep-old-snapshots"
```

Comment:
+1 for `--max-snapshots`, in the help text we can say "number of additional snapshots to keep" (or something like that), so that it should be clear that 0 doesn't mean none, just no additional.